sqe-api-server/Helpers/InputValidation.cs

```
@@ -18,6 +13,43 @@ namespace SQE.API.Server.Helpers
 {
     public static class GeometryValidation
     {
+        // You must declare the layout of your C struct (this is for any array)
+        [StructLayout(LayoutKind.Sequential)]
+        public struct BinaryData
+        {
+            public UIntPtr data;
+            public UIntPtr len;
+        }
+
+        // You must DllImport before each external function imported (beware the paths)
+        [DllImport(
+            "geo_repair_polygon",
+            CharSet = CharSet.Ansi,
+            CallingConvention = CallingConvention.Cdecl
+        )]
+        private static extern unsafe BinaryData repair_wkb(UIntPtr data, UIntPtr len);
+
+        [DllImport(
+            "geo_repair_polygon",
+            CharSet = CharSet.Ansi,
+            CallingConvention = CallingConvention.Cdecl
+        )]
+        private static extern unsafe void c_bin_data_free(BinaryData bin_data);
+
+        [DllImport(
+            "geo_repair_polygon",
+            CharSet = CharSet.Ansi,
+            CallingConvention = CallingConvention.Cdecl
+        )]
+        private static extern unsafe IntPtr repair_wkt(IntPtr wkt);
+
+        [DllImport(
+            "geo_repair_polygon",
+            CharSet = CharSet.Ansi,
```

Isn't RUST UTF-8 by default? It doesn't make any difference for the wkt polygons we move around, but still.

---

Yeah, but this travels over the C ABI, so ASCII is expected.  Any unicode characters would be invalid anyway, so it is fine to let it fail.